src/wcl/doc.h

```diff
@@ -62,10 +73,19 @@ struct doc_state {
   static doc_state from_string(const std::string& str) {
     doc_state state;
 
-    state.character_count = str.size();
+    state.byte_count = str.size();
+
+    const utf8proc_uint8_t* iter = reinterpret_cast<const unsigned char*>(str.c_str());
```

Comment:
The cast should be utf8proc_uint8_t not unsinged char. They're the same but we wan't to use the libraries type. The declared type can be `auto` for all I care.

Reply:
My bad. Fixed here https://github.com/sifive/wake/pull/969